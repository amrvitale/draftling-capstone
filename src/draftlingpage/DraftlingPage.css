.draftlingTitle {
    color: whitesmoke;
    font-size: 40px;
}