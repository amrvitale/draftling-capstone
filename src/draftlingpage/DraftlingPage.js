import React from 'react'
import ApiContext from '../ApiContext'
import config from '../config'
import Draftling from '../draftling/Draftling';
import {findDraftling} from '../draftlings-helpers'
import { Link } from 'react-router-dom'
import Edit from '../edit/Edit'

class DraftlingPage extends React.Component {
    static defaultProps = {
        match: {
          params: {}
        }
      }
      static contextType = ApiContext    
   
    
        

    render() {
        const {draftlings = [] } = this.context
        const {slug} = this.props.match.params
        console.log(slug)
        let selectedDraftling = draftlings.find(draftling => draftling.id === parseInt(slug))
        console.log(draftlings, selectedDraftling)
        console.log(selectedDraftling)


        return (
            <div>
              <section className="myDraftActions">
               <Link to={`/edit/${slug}`}>
                  <button type="button">Edit</button>
                </Link> 
                <button type="button">Publish</button>
                <button type="button">Delete</button>
              </section>
              <section className="draftViewOfSelected">
                <h1 className="draftlingTitle"> {(selectedDraftling) ? selectedDraftling.title: ""}</h1> 
                  <br />
                <p>Length Type: {(selectedDraftling) ? selectedDraftling.wordcount: ""}</p> 
                  <br />
                  <p>{(selectedDraftling) ? selectedDraftling.genre: ""}</p>
                  <br />
                <p>{(selectedDraftling) ? selectedDraftling.content: ""}</p>
              </section>

              
            </div>
        );
    }
}

export default DraftlingPage;