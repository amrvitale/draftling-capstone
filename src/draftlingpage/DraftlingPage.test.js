import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import DraftlingPage from './DraftlingPage'

describe (`DraftlingPage component`, () => {
    it('renders a .DraftlingPage by default', () => {
        const wrapper = shallow(<DraftlingPage />)
        expect (toJson(wrapper)).toMatchSnapshot()
    })

    
})