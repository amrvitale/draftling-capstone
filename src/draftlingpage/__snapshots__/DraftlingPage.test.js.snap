// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftlingPage component renders a .DraftlingPage by default 1`] = `
<div>
  <section
    className="myDraftActions"
  >
    <Link
      to="/edit/undefined"
    >
      <button
        type="button"
      >
        Edit
      </button>
    </Link>
    <button
      type="button"
    >
      Publish
    </button>
    <button
      type="button"
    >
      Delete
    </button>
  </section>
  <section
    className="draftViewOfSelected"
  >
    <h1
      className="draftlingTitle"
    >
       
    </h1>
    <br />
    <p>
      Length Type: 
    </p>
    <br />
    <p />
    <br />
    <p />
  </section>
</div>
`;
